docs/api/schema.ipynb

```diff
@@ -92,38 +143,179 @@
+   "metadata": {},
+   "source": [
+    "# Run the Clingo Solver\n",
+    "\n",
+    "The facts produced above act as constraints to the Clingo Solver. `run_clingo` calls the Clingo Solver on the program passed in and returns a list of models."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from draco import run_clingo"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "program = dict_to_facts(schema)\n",
+    "models = run_clingo(program)"
+   ]
+  },
+  {
+   "source": [
+    "`models` stores a list of models that satisfy the constraints passed to `run_clingo`"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "source": [
+    "# Converting Satisfiable Models back into Views\n",
+    "In order produce visualizations that satisfy the constraints, we must convert the answer sets of each model back to a nested data structure that resembles the schema. Models are objects of class `Model` which has members `answer_set` of type `clingo.Symbol`, `cost` of type of `int`, and `number` of type `int` "
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 23,
    "metadata": {},
    "outputs": [],
-   "source": []
+   "source": [
+    "from draco import answer_set_to_dict"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 24,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "execute_result",
+     "data": {
+      "text/plain": [
+       "[{'number_rows': 1461,\n",
+       "  'field': [{'name': 'date', 'type': 'datetime', 'unique': 1461},\n",
+       "   {'name': 'precipitation',\n",
+       "    'type': 'number',\n",
+       "    'unique': 111,\n",
+       "    'min': 0,\n",
+       "    'max': 55,\n",
+       "    'std': 6},\n",
+       "   {'name': 'temp_max',\n",
+       "    'type': 'number',\n",
+       "    'unique': 67,\n",
+       "    'min': -1,\n",
+       "    'max': 35,\n",
+       "    'std': 7},\n",
+       "   {'name': 'temp_min',\n",
+       "    'type': 'number',\n",
+       "    'unique': 55,\n",
+       "    'min': -7,\n",
+       "    'max': 18,\n",
+       "    'std': 5},\n",
+       "   {'name': 'wind',\n",
+       "    'type': 'number',\n",
+       "    'unique': 79,\n",
+       "    'min': 0,\n",
+       "    'max': 9,\n",
+       "    'std': 1},\n",
+       "   {'name': 'weather', 'type': 'string', 'unique': 5, 'freq': 714}]}]"
+      ]
+     },
+     "metadata": {},
+     "execution_count": 24
+    }
+   ],
+   "source": [
+    "views = [answer_set_to_dict(model.answer_set) for model in models]\n",
+    "views\n"
+   ]
+  },
+  {
+   "source": [
+    "In the example above, there is one satisfiable model which produces one nested data structure. This structure can be fed into visualization tools such as VegaLite to produce views. "
```

Comment:
VegaLite should always be spelled Vega-Lite.

There will always be only one model if there are no choice rules. As I said above, I don't think we want to demonstrate running clingo here.

It's also not correct that the structure above can be fed into Vega-Lite.